ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,55 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                context("message cannot be decoded or decrypted") {
+
+                    // RTL7e
+                    it("should deliver with encoding attribute set indicating the residual encoding and error should be emitted") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channelOptions = ARTChannelOptions(encrypted: ARTCrypto.defaultParams())
```

Comment:
No, I didn't. This is about constructors, which aren't specified in the API. We have `[ARTChannelOptions unencrypted]`, which makes an unencrypted `ChannelOptions`, and `[[ARTChannelOptions alloc] initEncrypted:params]`, which makes . The properties, `encrypted` and `cipherParams`, are as in the spec.

We probably should just change it to a single `initWithEncrypted:(bool) cipherParams:(ARTCipherParams *)` constructor. Can you make that change here, @ricardopereira? I will otherwise.

@mattheworiordan `ARTCrypto.defaultParams()` generates a random key. It's pretty stupid, yes, but it's for internal use, as in tests. `ARTCrypto` is not supposed to be part of the external API.